code/datums/personal_statistics.dm

```diff
@@ -0,0 +1,565 @@
+		if(receiver)
+			if(receiver_exists)	//Only record if they have a client
+				personal_statistics_receiver.chemicals_ingested[chem] += reagents_list[chem]
+		//If there is no receiver, that means the user is the one that needs their chems stat updated
+		else
+			personal_statistics_user.chemicals_ingested[chem] += reagents_list[chem]
+
+		//Determine if a healing chem was involved; only needs to be done once
+		if(!is_healing && istype(chem, /datum/reagent/medicine))
+			if(client_exists)
+				//If a receiving mob exists, we tally up to the user mob's stats that it performed a heal
+				if(receiver)
+					personal_statistics_user.heals++
+				else
+					personal_statistics_user.self_heals++
+			is_healing = TRUE
+	return TRUE
+
+///Determine if a self or non-self heal occurred, and tally up the user mob's respective stat
+/obj/item/stack/medical/proc/record_healing(mob/living/user, mob/living/receiver)
+	if(!user.client)
+		return FALSE
+
+	var/datum/personal_statistics/personal_statistics_user = GLOB.personal_statistics_list[user.client]
+	if(user == receiver)
+		receiver = null
+
+	//If a receiving mob exists, we tally up to the user mob's stats that it performed a heal
+	if(receiver)
+		personal_statistics_user.heals++
+	else
+		personal_statistics_user.self_heals++
+	return TRUE
+
+///Record what was drank and if it was medicinal
+/obj/machinery/deployable/reagent_tank/proc/record_sippies(total, list/reagents_list, mob/user)
+	if(!total || !LAZYLEN(reagents_list) || !user.client)
+		return FALSE
+
+	var/is_healing
+	var/datum/personal_statistics/personal_statistics = GLOB.personal_statistics_list[user.client]
+	for(var/chem in reagents_list)
+		//Add the chem and amount consumed to the list
+		personal_statistics.chemicals_ingested[chem] += reagents_list[chem]
+		//Determine if a healing chem was involved; only needs to be done once
+		if(!is_healing && istype(chem, /datum/reagent/medicine))
+			personal_statistics.self_heals++
+			is_healing = TRUE
+	personal_statistics.sippies++
+	return TRUE
+
+///Tally up the corresponding weapon used by the pilot into their /datum/personal_statistics
+/obj/docking_port/mobile/marine_dropship/casplane/proc/record_cas_activity(obj/structure/dropship_equipment/weapon/weapon)
+	if(!chair.occupant.client)
+		return FALSE
+
+	var/datum/personal_statistics/personal_statistics = GLOB.personal_statistics_list[chair.occupant.client]
+	//Determine the type path of the weapon used and then tally accordingly
+	switch(weapon.type)
+		if(/obj/structure/dropship_equipment/weapon/heavygun)
+			personal_statistics.cas_cannon_shots++
+		if(/obj/structure/dropship_equipment/weapon/laser_beam_gun)
+			personal_statistics.cas_laser_shots++
+		if(/obj/structure/dropship_equipment/weapon/minirocket_pod)
+			personal_statistics.cas_minirockets_fired++
+		if(/obj/structure/dropship_equipment/weapon/rocket_pod)
+			personal_statistics.cas_rockets_fired++
```

Comment:
Cant you automate this
ex:
list(name = timesfired)  -> "Name was fired timesfired times"
instead of this snowflake switch on .type